servers/rendering/renderer_rd/renderer_scene_render_rd.cpp

```diff
@@ -3413,45 +3413,45 @@ void RendererSceneRenderRD::_setup_lights(const PagedArray<RID> &p_lights, const
 					continue;
 				}
 
-				const real_t distance = camera_plane.distance_to(li->transform.origin);
+				const real_t depth = camera_plane.distance_to(li->transform.origin);
 
 				if (storage->light_is_distance_fade_enabled(li->light)) {
+					// Use `distance_squared_to()` to speed up distance checks.
+					const real_t distance = p_camera_transform.origin.distance_squared_to(li->transform.origin);
 					const float fade_begin = storage->light_get_distance_fade_begin(li->light);
 					const float fade_length = storage->light_get_distance_fade_length(li->light);
 
-					if (distance > fade_begin) {
-						if (distance > fade_begin + fade_length) {
-							// Out of range, don't draw this light to improve performance.
-							continue;
-						}
+					if (distance > Math::pow(fade_begin + fade_length, 2)) {
+						// Out of range, don't draw this light to improve performance.
+						continue;
 					}
 				}
 
 				cluster.omni_light_sort[cluster.omni_light_count].instance = li;
-				cluster.omni_light_sort[cluster.omni_light_count].depth = distance;
+				cluster.omni_light_sort[cluster.omni_light_count].depth = depth;
```

Comment:
Is there a compelling reason to use camera depth here instead of just using distance everywhere?

Reply:
I don't know if the clustering code expects the distance to be "flat" (relative to the camera frustum). I kept it this way to avoid regressions.